pook/interceptors/websocket.py

```diff
@@ -0,0 +1,115 @@
+from ..request import Request
+from .base import BaseInterceptor
+
+# Support Python 2/3
+try:
+    import mock
+except Exception:
+    from unittest import mock
+
+
+class MockFrame(object):
+    def __init__(self, opcode, data, fin=1):
+        self.opcode = opcode
+        self.data = data
+        self.fin = fin
+
+
+class MockSocket(object):
+    def fileno(self):
+        return 0
+
+    def gettimeout(self):
+        return 0
+
+
+class MockHandshakeResponse(object):
+
+    def __init__(self, status, headers, subprotocol):
+        self.status = status
+        self.headers = headers
+        self.subprotocol = subprotocol
+
+
+class WebSocketInterceptor(BaseInterceptor):
+
+    def _handshake(self, sock, hostname, port, resource, **options):
+        return MockHandshakeResponse(200, {}, "")
+
+    def _connect(self, url, options, proxy, socket):
+        req = Request()
+        req.headers = {}  # TODO
+        req.url = url
+
+        # TODO does this work multithreaded?!?
+        self._mock = self.engine.match(req)
```

Comment:
It is not. Mutex synchronization is required to make it safe. However, I feel that's another feature that has to be implemented in a separate PR.